packages/flutter/test/painting/fractional_offset_test.dart

```diff
@@ -98,6 +98,35 @@ void main() {
     expect(FractionalOffsetGeometry.lerp(mixed1, mixed2, 0.25), mixed3);
   });
 
+  test('lerp commutes with resolve', () {
+    final List<FractionalOffsetGeometry> offsets = <FractionalOffsetGeometry>[
+      const FractionalOffset(-1.0, 0.65),
+      const FractionalOffsetDirectional(-1.0, 0.45),
+      const FractionalOffsetDirectional(0.125, 0.625),
+      const FractionalOffset(0.25, 0.875),
+      const FractionalOffset(0.0625, 0.5625).add(const FractionalOffsetDirectional(0.1875, 0.6875)),
+      const FractionalOffsetDirectional(2.0, 3.0),
+      const FractionalOffset(2.0, 3.0),
+      const FractionalOffset(2.0, 3.0).add(const FractionalOffsetDirectional(5.0, 3.0)),
+      const FractionalOffset(10.0, 20.0).add(const FractionalOffsetDirectional(30.0, 50.0)),
+      const FractionalOffset(70.0, 110.0).add(const FractionalOffsetDirectional(130.0, 170.0)),
+      const FractionalOffset(25.0, 42.5).add(const FractionalOffsetDirectional(55.0, 80.0)),
```

Comment:
maybe add null and use elvis in the for loop?

Reply:
I'll do that in the patch where I change the semantics of null.